### PR TITLE
perf(rum): move to ES6 modules to reduce bundle size

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ part of elastic/apm-agent-rum-js#159 
+ Need to transform node test files for Integration tests before jasmine

#### Savings - gzipped
Filename | Before(master) | After 
-------- | --------| ---------
elastic-apm-js-base.min.js |  16.04  | **15.75**

### CODE REVIEW

File: packages/rum/src/index.js [hunk 1]
```diff
@@ -23,27 +23,19 @@
...
-  default: apmBase.init.bind(apmBase),
-  init: apmBase.init.bind(apmBase),
-  ApmBase,
-  apmBase,
-  apm: apmBase
-}
+const init = () => apmBase.init.bind(apmBase)
 
-module.exports = exports
+export default apmBase
```
Comment (jahtalab) [inline packages/rum/src/index.js]: The previous default was the `init` method on `apmBase` and not the `apmBase` itself.

Comment (vigneshshanmugam) [reply packages/rum/src/index.js]: Good catch 👍

File: packages/rum-core/src/index.js [hunk 2]
```diff
@@ -25,21 +25,24 @@
...
+  return serviceFactory
+}
+
+export {
+  createServiceFactory,
   ServiceFactory,
-  patching,
-  utils
+  patchAll,
+  subscription,
```
[the remark this answers is not in the document]
Comment (vigneshshanmugam) [reply packages/rum-core/src/index.js]: changed to patchSubscription as discussed.

File: packages/rum/src/index.js [hunk 3]
```diff
@@ -23,27 +23,19 @@
...
 
-var exports = {
-  __esModule: true,
-  default: apmBase.init.bind(apmBase),
-  init: apmBase.init.bind(apmBase),
-  ApmBase,
-  apmBase,
-  apm: apmBase
-}
+const init = () => apmBase.init.bind(apmBase)
```
Comment (jahtalab) [inline packages/rum/src/index.js]: The init is defined as a function that returns a function which is the actual init function. instead it should just be the actual init function

**Comment by vigneshshanmugam** [comment]
@jahtalab Check last 2 commits where I have added tests for exports and added support for transplred modules for Node >=8 version.

**Comment by vigneshshanmugam** [comment]
Test failures are unrelated to the PR. It fails on Chrome 49 which is a known race condition bug.

File: packages/rum-core/package.json [hunk 4]
```diff
@@ -13,15 +15,21 @@
     "directory": "packages/rum-core"
   },
   "scripts": {
+    "prepublish": "npm run build:main",
+    "build:main": "npx babel src -d lib --config-file ../../babel.node.js",
```
Comment (jahtalab) [inline packages/rum-core/package.json]: Maybe we should put this under `dist/lib` as well, WDYT?

File: packages/rum/package.json [hunk 5]
```diff
@@ -2,7 +2,9 @@
   "name": "@elastic/apm-rum",
   "version": "4.0.1",
   "description": "Elastic APM JavaScript agent",
-  "main": "src/index.js",
+  "main": "dist/lib/index.js",
+  "module": "src/index.js",
+  "sideEffects": false,
```
Comment (jahtalab) [inline packages/rum/package.json]: Doesn't patching browser API counts as sideEffects?

File: packages/rum/run-script.js [hunk 6]
```diff
@@ -113,26 +113,26 @@ function serveE2e(servingPath, port) {
...
       // The stack is not useful in this context.
       err.stack = ''
       cb(err)
     } else {
       cb()
     }
   })
-
-  jrunner.showColors(true)
-  jrunner.addReporter(JasmineRunner.ConsoleReporter(jrunner))
```
Comment (jahtalab) [inline packages/rum/run-script.js]: Just to make sure this is intentional, we don't need to add the console report manually?

File: packages/rum-core/package.json (hunk 4 above)
Comment (vigneshshanmugam) [reply packages/rum-core/package.json]: sure, lets do it.

File: packages/rum-core/package.json [hunk 7]
```diff
@@ -13,15 +15,21 @@
...
     "karma": "karma start",
     "karma:bench": "karma start karma.bench.conf.js",
     "karma:coverage": "karma start --coverage",
     "test:unit": "node ../../dev-utils/run-script.js runUnitTests true packages/rum-core",
     "test": "npm run test:unit"
   },
   "files": [
-    "src"
+    "src",
+    "lib"
```
Comment (jahtalab) [inline packages/rum-core/package.json]: I think this is not needed anymore.

File: .gitignore [hunk 8]
```diff
@@ -15,3 +15,5 @@ coverage
 
 # ignore lerna-debug and npm-debug logs
 *debug.log
+
+/packages/*/lib
```
Comment (jahtalab) [inline .gitignore]: This is not needed anymore, right?

File: packages/rum-core/package.json (hunk 7 above)
Comment (jahtalab) [reply packages/rum-core/package.json]: It actually needs to be the dist folder.

Comment (vigneshshanmugam) [reply packages/rum-core/package.json]: Oops yeah, changing it now

**Pull request merged**